modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -373,10 +400,38 @@ private PredicateImpl(Scalar scalar) {
         }
     }
 
+    /** */
+    private class BiPredicateImpl implements BiPredicate<Row, Row> {
```

Comment:
Can we use common parent to deduplicate code?

Reply:
done, thanks !

Reply:
But I don't see any changes here.